homeassistant/components/apple_tv/media_player.py

```diff
@@ -48,49 +49,53 @@
 )
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Set up the Apple TV platform."""
-    if not discovery_info:
-        return
-
-    # Manage entity cache for service handler
-    if DATA_ENTITIES not in hass.data:
-        hass.data[DATA_ENTITIES] = []
-
-    name = discovery_info[CONF_NAME]
-    host = discovery_info[CONF_HOST]
-    atv = hass.data[DATA_APPLE_TV][host][ATTR_ATV]
-    power = hass.data[DATA_APPLE_TV][host][ATTR_POWER]
-    entity = AppleTvDevice(atv, name, power)
-
-    @callback
-    def on_hass_stop(event):
-        """Stop push updates when hass stops."""
-        atv.push_updater.stop()
-
-    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, on_hass_stop)
-
-    if entity not in hass.data[DATA_ENTITIES]:
-        hass.data[DATA_ENTITIES].append(entity)
-
-    async_add_entities([entity])
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Load Apple TV media player based on a config entry."""
+    name = config_entry.data[CONF_NAME]
+    manager = hass.data[DOMAIN][config_entry.unique_id]
+    async_add_entities([AppleTvDevice(name, config_entry.unique_id, manager)])
 
 
 class AppleTvDevice(MediaPlayerEntity):
     """Representation of an Apple TV device."""
 
-    def __init__(self, atv, name, power):
+    def __init__(self, name, identifier, manager):
         """Initialize the Apple TV device."""
-        self.atv = atv
+        self.atv = None
         self._name = name
+        self._identifier = identifier
         self._playing = None
-        self._power = power
-        self._power.listeners.append(self)
-        self.atv.push_updater.listener = self
+        self._manager = manager
 
     async def async_added_to_hass(self):
         """Handle when an entity is about to be added to Home Assistant."""
-        self._power.init()
+        self._manager.listeners.append(self)
```

Comment:
This is not allowed. Connect the dispatch helper here instead.

Reply:
Will fix that with dispatcher.